aiida/manage/tests/pytest_fixtures.py

```diff
@@ -42,27 +61,243 @@ def aiida_caplog(caplog):
     AIIDA_LOGGER.propagate = propogate
 
 
-@pytest.fixture(scope='session', autouse=True)
-def aiida_profile():
-    """Set up AiiDA test profile for the duration of the tests.
+@pytest.fixture(scope='session')
+def postgres_cluster(
```

Comment:
it looks to me like some of the code below duplicates what is already in the `TemporaryProfileManage` in `main.py`. Can we avoid that?

Reply:
Not in a way that I can see that will make things more clear. The problem of the `TemporaryProfileManager` is that it doesn't properly separate concerns but mixes various concepts: The concept of a profile, but hard-codes the integration with a disk repository and a PostgreSQL cluster for a database. The whole point is to properly decouple this so that parts can be easily replaced where needed. I think it is better to deprecate the entire module as I have done, then try and keep bits around and make things complicated. I would have even removed it entirely instead of deprecating it, because the main use case is for plugin to use `aiida_profile`, but I am not sure if code out there used these classes directly.